Applications/Terminal/Terminal.cpp

```diff
@@ -837,52 +972,37 @@ void Terminal::paint_event(GPaintEvent& event)
 
     GPainter painter(*this);
 
-    if (m_needs_background_fill) {
-        m_needs_background_fill = false;
-        if (m_visual_beep_timer.is_active())
-            painter.fill_rect(frame_inner_rect(), Color::Red);
-        else
-            painter.fill_rect(frame_inner_rect(), Color(Color::Black).with_alpha(255 * m_opacity));
-    }
-
-    if (m_rows_to_scroll_backing_store && m_rows_to_scroll_backing_store < m_rows) {
-        int first_scanline = m_inset;
-        int second_scanline = m_inset + (m_rows_to_scroll_backing_store * m_line_height);
-        int num_rows_to_memcpy = m_rows - m_rows_to_scroll_backing_store;
-        int scanlines_to_copy = (num_rows_to_memcpy * m_line_height) - m_line_spacing;
-        memcpy(
-            painter.target()->scanline(first_scanline),
-            painter.target()->scanline(second_scanline),
-            scanlines_to_copy * painter.target()->pitch()
-        );
-        line(max(0, m_cursor_row - m_rows_to_scroll_backing_store)).dirty = true;
-    }
-    m_rows_to_scroll_backing_store = 0;
-
+    if (m_visual_beep_timer.is_active())
```

Comment:
Hrm, full red is going to be a little bit retina-melting. Maybe it should be a dark red, or something? Probably OK at least for now.

Reply:
Yeah that makes sense lol, but we already merged the PR that added/uses this color. Take it up with @awesomekling if you want a different design decision, he seemed happy with it.

Reply:
Right, I didn't see that it was already there.

Reply:
Yeah, the red is definitely a much and we should improve on it. Let's keep the scope of each PR tight though.

Reply:
Yep, I wouldn't have commented if I had read the left hand side of the diff closely. I didn't realise it had already been added, it was a nice new feature to me. ;-)